### PR TITLE
feat(core): error when running atomized tasks outside of DTE

### PR DESCRIPTION
**Pull request by MaxKless**
## Current Behavior
Nx allows running atomized tasks (like `e2e-ci`) anywhere.

## Expected Behavior
Nx allows running atomized tasks (like `e2e-ci`) only in distributed environments like Nx Agents or DTE. We'll leave that actual check to nx cloud but in the default runner, we'll throw an error.
It's possible to escape from this with an env var.

### CODE REVIEW

File: packages/nx/src/tasks-runner/task-graph-utils.ts [hunk 1]
```diff
@@ -66,3 +70,42 @@ export function makeAcyclic(graph: {
...
+  projectGraph: ProjectGraph
+): void {
+  if (process.env['NX_SKIP_ATOMIZER_VALIDATION']) {
+    return;
+  }
+  const tasksWithAtomizer = Object.values(taskGraph.tasks).filter(
+    (task) =>
+      projectGraph.nodes[task.target.project]?.data?.targets?.[
+        task.target.target
+      ]?.metadata?.nonAtomizedTarget !== undefined
```
Comment (FrozenPandaz) [inline packages/nx/src/tasks-runner/task-graph-utils.ts:85]: Mm, not sure I like this. `metadata` is not supposed to affect running of tasks :thinking:

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/nx/src/tasks-runner/task-orchestrator.ts [hunk 2]
```diff
@@ -388,7 +388,15 @@ export class TaskOrchestrator {
         task,
         this.projectGraph
       );
-      if (
+
+      if (targetConfiguration.executor === 'nx:noop') {
```
Comment (FrozenPandaz) [inline packages/nx/src/tasks-runner/task-orchestrator.ts:392]: Was this important?

File: packages/nx/src/tasks-runner/run-command.ts [hunk 3]
```diff
@@ -129,6 +133,14 @@ function createTaskGraphAndValidateCycles(
     }
   }
 
+  // validate that no atomized tasks like e2e-ci are used without Nx Cloud
+  if (
+    !isNxCloudUsed(readNxJson()) &&
```
Comment (FrozenPandaz) [inline packages/nx/src/tasks-runner/run-command.ts:138]: Mm this makes sense

File: docs/nx-cloud/features/split-e2e-tasks.md [hunk 4]
```diff
@@ -34,6 +34,13 @@ nx add @nx/playwright
 nx add @nx/jest
 ```
 
+{% /tab %}
+{% tab label="Gradle" %}
+
+```shell {% skipRescope=true %}
+nx add @nx/gradle
+```
+
```
Comment (FrozenPandaz) [inline docs/nx-cloud/features/split-e2e-tasks.md:43]: Can we leave out the gradle changes until we've tested it on Ocean and ensure that it's production ready?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
Looking good! :+1: Few more changes and we should be good to go.

File: packages/nx/src/tasks-runner/task-orchestrator.ts (hunk 2 above)
Comment (MaxKless) [reply packages/nx/src/tasks-runner/task-orchestrator.ts:392]: bit of a leftover from when I was working there but it makes more sense to me to handle special cases first and then the general cases... I can remove it though.

Comment (MaxKless) [reply packages/nx/src/tasks-runner/task-orchestrator.ts]: I reverted the change anyways

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**